### ISSUE TITLE
tree-sitter-elm.wasm not found

### ISSUE THREAD

**Issue by jweir** [elm-tooling/elm-language-server#1052]
Starting vim with elm-language-server 2.7.1 I get the following error:
```
[coc.nvim] Request initialize failed with message: ENOENT: no such file or directory, open '../../../../../../opt/homebrew/lib/node_modules/@elm-tooling/
[coc.nvim] Server languageserver.elmLS failed to start: Error: Request initialize failed with message: ENOENT: no such file or directory, open '../../../
../../../opt/homebrew/lib/node_modules/@elm-tooling/elm-language-server/out/common/tree-sitter-elm.wasm'
```

## Current Workaround

I adjusted the path to tree-sitter-elm to the below, but this probably isn't a great solution.

```
        const pathToWasm = (_b = initializationOptions.treeSitterElmWasmUri) !== null && _b !== void 0 ? _b : Path.relative(process.cwd(), Path.join(__dirname, '../../out/', "tree-sitter-elm.wasm"));
```

## Your Environment

* Version used: 2.7.1
* Editor used: vim + coc 
* Environment name and version (e.g. node.js 5.4): node v18.16.1
* Operating System and version: MacOS 13.4.1

**Comment by razzeee** [comment]
Could be this change https://github.com/elm-tooling/elm-language-server/commit/1dd8d658a7ac79e50136d98242abb27af6ee87c6#diff-1f963b5ae3613de37b5354a3da594cdef878e0715ad0eb3896bd701e2511b273R60

**Comment by Chocrates** [comment]
Same issue here with a relatively new version of NVIM.  Using Packer and Mason and did a `npm install -g @elm-tooling/elm-language-server` though not sure that was needed.
Gonna try the workaround.
Where did you put this workaround @jweir?

**Comment by jweir** [comment]
@Chocrates  I edited `out/common/index.js` line 74 under `/opt/homebrew/lib/node_modules/@elm-tooling/elm-language-server/` – your location may vary.

**Comment by Chocrates** [comment]
Thanks that worked for me @jweir

**Comment by dtrckd** [comment]
Same error here. I paste the stack I got in neovim as it is slightly different from @jweir, in case it could helps

 ``
Error executing vim.schedule lua callback: /snap/nvim/2809/usr/share/nvim/runtime/lua/vim/lsp.lua:1309: RPC[Error] code_name = InternalError, message = "Request initialize failed with message: ENOENT: no such file
 or directory, open '../../../../../.nvm/versions/node/v18.4.0/lib/node_modules/@elm-tooling/elm-language-server/out/common/tree-sitter-elm.wasm'"                                                                   
stack traceback:                                                                                                                                                                                                     
        [C]: in function 'assert'                                                                                                                                                                                    
        /snap/nvim/2809/usr/share/nvim/runtime/lua/vim/lsp.lua:1309: in function ''                                                                                                                                  
        vim/_editor.lua: in function <vim/_editor.lua:0> `
```

**Comment by aaaaargZombies** [comment]
Just ran into same problem in Neovim using Mason for LSP business. 

```
Error executing vim.schedule lua callback: /usr/share/nvim/runtime/lua/vim/lsp.lua:1309: RPC[Error] code_name = InternalError, message = "Request initialize fa
iled with message: ENOENT: no such file or directory, open '../../.local/share/nvim/mason/packages/elm-language-server/node_modules/@elm-tooling/elm-language-s
erver/out/common/tree-sitter-elm.wasm'"                                                                                                                        
stack traceback:                                                                                                                                               
        [C]: in function 'assert'                                                                                                                              
        /usr/share/nvim/runtime/lua/vim/lsp.lua:1309: in function ''                                                                                           
        vim/_editor.lua: in function <vim/_editor.lua:0>
```

I've just downgrading the package rather than messing with the source

`:MasonInstall elm-language-server@2.7.0`

**Comment by seanstrom** [comment]
I think this issue is caused by a recent refactor of the code into separate folders.  For example this section of code used to be in the `src` directory but was refactored into the `src/common` directory, which likely silently changed the computed directory for the `tree-sitter-elm.wasm` file: https://github.com/elm-tooling/elm-language-server/blob/ddab0447c7c32604f1fcf4ac3ac40e0b3dfe121a/src/common/index.ts#L60-L65

At the moment, I've shared a potential fix at #1070 
Please let me know what you think 🙏

**Comment by jmbockhorst** [comment]
Fixed by #1070.

**Comment by razzeee** [comment]
This needs a new server release to reach vim users
